build/ris.js

```diff
@@ -0,0 +1,397 @@
+'use strict'
+
+import https from 'https';
+import fs from 'node:fs';
+import {PassThrough} from 'node:stream'
+import {pipeline} from 'node:stream'
+import ndjson from 'ndjson'
+import {stringify} from 'qs'
+
+const bbox = { // should be approx. quadratic
+    north: 70.0,
+    west: -10.0,
+    south: 35.0,
+    east: 40.0,
+}
+
+const LIMIT_RESULTS = 10000;
+const PADDING_RATIO = 1.1;
+const MAX_DEPTH = 8;
+const MAX_REQUESTS = 1000;
+const TMP_PATH = 'tmp/';
+const STADA_FILE = 'stada.json'
+
+const DEFAULT_PRODUCTS = {
+    nationalExpress: false,
+    national: false,
+    regionalExpress: false,
+    regional: false,
+    suburban: false,
+    subway: false,
+    tram: false,
+    bus: false,
+    taxi: false,
+    ferry: false
+};
+
+const RIS_PRODUCTS_MAPPING = {
+    HIGH_SPEED_TRAIN: {product: 'nationalExpress', weight: 3},
+    INTERCITY_TRAIN: {product: 'national', weight: 2},
+    INTER_REGIONAL_TRAIN: {product: 'regionalExpress', weight: 1},
+    REGIONAL_TRAIN: {product: 'regional', weight: 1},
+    CITY_TRAIN: {product: 'suburban', weight: 1},
+    SUBWAY: {product: 'subway', weight: 0.5},
+    TRAM: {product: 'tram', weight: 0.2},
+    BUS: {product: 'bus', weight: 0.1},
+    TAXI: {product: 'taxi', weight: 0.1},
+    FERRY: {product: 'ferry', weight: 0.2},
+    SHUTTLE: {product: 'taxi', weight: 0.1}
+}
+
+const MEMBER_WEIGHT = 0.2;
+const LOWEST_PRICE_CATEGORY = 8;
+const PRICE_CATEGORY_WEIGHT = 0.5;
+
+const DAY_MAPPING = {
+    "monday": 'Mo',
+    "tuesday": 'Di',
+    "wednesday": 'Mi',
+    "thursday": 'Do',
+    "friday": 'Fr',
+    "saturday": 'Sa',
+    "sunday": 'So',
+    "holiday": 'Feiertag'
+}
+
+const abortWithError = (err) => {
+    console.error(err)
+    process.exit(1)
+}
+
+const data = new PassThrough({
+    objectMode: true,
+})
+
+function sleep(ms) {
+    return new Promise((resolve) => {
+        setTimeout(resolve, ms);
+    });
+}
+
+const exists = async (f) => {
+    try {
+        await fs.promises.stat(f);
+        return true;
+    } catch (_) {
+        return false;
+    }
+}
+
+const doRequest = (options, data) => {
```

Comment:
Let's clarify that it parses JSON.

```suggestion
const requestJson = async (options, data) => {
```